### PR TITLE
Backend snapshot tests: separate snapshot for each major version

### PR DESCRIPTION
**Pull request by NachoSoto**
Fixes [CF-547].

I recommend reviewing this ignoring JSON files:https://github.com/RevenueCat/purchases-ios/pull/1504/files?file-filters%5B%5D=.pbxproj&file-filters%5B%5D=.swift&show-deleted-files=true&show-viewed-files=true

This also combines the duplicate copies of some tests for each version, as only 1 is needed now. `assertSnapshot` encodes the version in the filename.
Combined:
- `BackendPostRecepitDataTests`
- `BackendSubscriberAttributesTests`

No more unnecessary skipped tests.
- Before: `Number of tests skipped | 112`
- Now: `Number of tests skipped | 58`

### Example:
Screen Shot 2022-04-13 at 16 07 15

[CF-547]: https://revenuecats.atlassian.net/browse/CF-547?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### ⚠️ Note:

Because iOS 12.x simulators currently crash on M1 Macs, I haven't generated those snapshots yet.

### CODE REVIEW

File: Tests/UnitTests/TestHelpers/OSVersionEquivalent.swift [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                let majorVersion = UIDevice.majorVersion
+
+                if let equivalent = Self(rawValue: majorVersion) {
+                    return equivalent
+                } else {
+                    throw Error.unknownOS()
+                }
+            #endif
+        }
+    }
```
Comment (NachoSoto) [inline Tests/UnitTests/TestHelpers/OSVersionEquivalent.swift:52]: This is the core of this PR. Thanks @taquitos for the idea.

File: Tests/UnitTests/TestHelpers/OSVersionEquivalent.swift [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+enum OSVersionEquivalent: Int {
+
+    case iOS12 = 12
+    case iOS13 = 13
+    case iOS14 = 14
+    case iOS15 = 15
+
+}
+
+extension OSVersionEquivalent {
```
Comment (taquitos) [inline Tests/UnitTests/TestHelpers/OSVersionEquivalent.swift:31]: 🕺

**Review by taquitos — CHANGES_REQUESTED** [review]
Awesome stuff, one possible change I see some files prefixed like `iOSiOS15`

**Comment by NachoSoto** [comment]
Oops good catch! I forgot we have those separate tests as well.

**Comment by NachoSoto** [comment]
Combined `BackendSubscriberAttributesTests` as well and updated snapshots.

File: Tests/UnitTests/Networking/Backend/BackendPostReceiptDataTests.swift [hunk 3]
```diff
@@ -7,7 +7,7 @@
 //
 //      https://opensource.org/licenses/MIT
 //
-//  BackendPostReceiptDataTestBase.swift
+//  BackendPostReceiptDataTest.swift
```
Comment (taquitos) [inline Tests/UnitTests/Networking/Backend/BackendPostReceiptDataTests.swift:10]: `BackendPostReceiptDataTest` -> `BackendPostReceiptDataTests`

**Review by taquitos — APPROVED** [review]
The smallest comment update, but then 🚢

File: Tests/UnitTests/Networking/Backend/BackendPostReceiptDataTests.swift (hunk 3 above)
Comment (NachoSoto) [reply Tests/UnitTests/Networking/Backend/BackendPostReceiptDataTests.swift:10]: Oops thanks

**Comment by NachoSoto** [comment]
I'll wait to merge this until somebody can generate the iOS 12 snapshots for me 🙏🏻

**Comment by taquitos** [comment]
> I'll wait to merge this until somebody can generate the iOS 12 snapshots for me 🙏🏻

updating my Intel mac 🙃

**Comment by taquitos** [comment]
@NachoSoto this is ready when you are 😄

**Comment by NachoSoto** [comment]
Awesome thanks! Just pushed the comment fix, I'll merge this and make sure they're passing in `main`.

**Pull request merged**